rust/src/download.rs

```diff
@@ -0,0 +1,62 @@
+use crate::bs_safe::{BridgeStanError, Result};
+use crate::VERSION;
+use flate2::read::GzDecoder;
+use log::info;
+use std::{env::temp_dir, fs, path::PathBuf};
+use tar::Archive;
+
+/// Download and unzip the BridgeStan source distribution for this version
+/// to ~/.bridgestan/bridgestan-version
+pub fn download_bridgestan_src() -> Result<PathBuf> {
+    let homedir = dirs::home_dir().unwrap_or(temp_dir());
+
+    let bs_path_download_temp = homedir.join(".bridgestan_tmp_dir");
+    let bs_path_download = homedir.join(".bridgestan");
```

Comment:
[optional, if not in this PR I will circle back around to do it eventually] It would be nice to have a similar warning to what we do in Python if a lot of these exist:

https://github.com/roualdes/bridgestan/blob/ef1f98694b9c6236e2d0b96eba025e864150e2de/python/bridgestan/compile.py#L73-L78